apps/ledger-live-mobile/src/screens/ReceiveSelectAccount.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useMemo } from "react";
+import { useSelector } from "react-redux";
+import { useTranslation } from "react-i18next";
+import { FlatList } from "react-native";
+import {
+  accountWithMandatoryTokens,
+  flattenAccounts,
+} from "@ledgerhq/live-common/lib/account/helpers";
+import { CryptoCurrency, TokenCurrency } from "@ledgerhq/live-common/lib/types";
+import { Flex, Text } from "@ledgerhq/native-ui";
+import { ScreenName } from "../const";
+import { accountsSelector } from "../reducers/accounts";
+import AccountCard from "../components/AccountCard";
+import type { Account, AccountLike } from "@ledgerhq/live-common/lib/types/account";
+
+type Props = {
+  navigation: any;
+  route: {
+    params: {
+      selectedCurrency: CryptoCurrency | TokenCurrency;
+    };
+  };
+};
+
+export default function SelectAccount({ navigation, route }: Props) {
+  const { selectedCurrency } = route.params;
+
+  const { t } = useTranslation();
+  const allAccounts = useSelector(accountsSelector);
+  const currencyAccounts = useMemo(() => {
+    if (selectedCurrency) {
+      const filteredAccounts = allAccounts.filter(
+        acc =>
+          acc.currency.id ===
+          (selectedCurrency.type === "TokenCurrency"
+            ? selectedCurrency.parentCurrency.id
+            : selectedCurrency.id),
+      );
+      if (selectedCurrency.type === "TokenCurrency") {
+        // add in the token subAccount if it does not exist
+        return flattenAccounts(
+          filteredAccounts.map(acc =>
+            accountWithMandatoryTokens(acc, [selectedCurrency]),
+          ),
+        ).filter(
+          acc =>
+            acc.type === "Account" ||
+            (acc.type === "TokenAccount" &&
+              acc.token.id === selectedCurrency.id),
+        );
+      }
+      return flattenAccounts(filteredAccounts);
+    }
+    return flattenAccounts(allAccounts);
+  }, [allAccounts, selectedCurrency]);
```

Comment:
the logic inside that "filter accounts for a currency" would be beneficial to be made a live-common utility in future **with unit tests** & possibly reused by LLD :)

Reply:
Totally agree